Protocol.md

```diff
@@ -23,188 +45,183 @@ The following figure gives a brief overview of the protocol:
 
-1. Funding Change Address
-   - The address to use to send the change if the sum of A's inputs is greater than the fund amount plus the fees.
-2. Final Address
-   - The address to which funds will be sent in [unilateral contract execution](Transactions.md#CETOutputs), [refund](Transactions.md#refund-transaction) and [mutual closing](Transactions.md#mutual-closing-transaction) transactions.
+`contract_maturity_bound` is the nLockTime to be put on CETs. `contract_timeout` is the nLockTime to be put on the refund transaction.
 
-#### A's inputs
-The set of UTXOs to be used as input to the fund transaction.
+#### Requirements
 
-#### Estimatesmartfee (satoshi/vbyte)
-The fee rate to be used when computing fees for the different transactions.
+The sending node MUST:
 
-#### CET CSV Delay
-The number to use as input to the CSV in the [CETs](Transactions.md#outputs-1).
+  - set undefined bits in `contract_flags` to 0.
+  - ensure the `chain_hash` value identifies the chain it wishes to open the contract within.
+  - set `funding_pubkey` to a valid secp256k1 pubkey in compressed format.
+  - set `total_collateral_satoshis` to a value greater than or equal to 1000.
+  - set `contract_maturity_bound` and `contract_timeout` to either both be UNIX timestamps, or both be block heights as distinguished in [BIP 65](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki).
+  - set `contract_maturity_bound` to be less than `contract_timeout`.
 
-#### CET lock time
-The number to use as `nLockTime` for the CET transactions.
-This should correspond to the maturity date of the contracts, invalidating early CET broadcasting.
+The sending node SHOULD:
 
-#### Refund locktime
-The locktime to be used for the refund transaction.
-It should be set at a later date than the maturity date of the contract.
+  - set `feerate_per_kw` to at least the rate it estimates would cause the transaction to be immediately included in a block.
+  - set `contract_maturity_bound` to the earliest expected oracle signature time.
+  - set `contract_timeout` sufficiently long after `contract_maturity_bound` to allow for late oracle signatures and other delays to closing the contract.
 
----
+The receiving node MUST:
 
-### Accept
-After receiving the `offer` message from A, and after [validating](#offer-validation) the provided information, B creates and sends back an `accept` message containing:
-1. B's fund amount (satoshi)
-1. B's public keys
-1. B's addresses
-1. B's inputs
-1. CET signatures
-1. Refund signature
+  - ignore undefined bits in `channel_flags`.
 
-#### B's fund amount (satoshi)
-How much B inputs into the contract.
+The receiving node MAY reject the contract if:
 
-#### B's public keys
-The following public keys:
+  - it does not agree to the terms in `contract_info`.
+  - the `contract_info` is missing relevant events.
+  - it does not want to use the oracle(s) specified in `oracle_info`.
+  - `total_collateral_satoshis` is too small.
 
```

Comment:
Above you say the sending node must

> set `feerate_per_kw` to at least the rate it estimates would cause the transaction to be immediately included in a block

Should tell the receiving node to verify this